test/Generics/same_type_constraints.swift

```diff
@@ -351,11 +351,11 @@ func intercomponentSameComponents<T: P10>(_: T)
         T.B == T.A { } // expected-note{{previous same-type constraint 'T.B' == 'T.A' written here}}
 
 func intercomponentMoreThanSpanningTree<T: P10>(_: T)
-  where T.A == T.B,
+  where T.A == T.B, // expected-note {{previous same-type constraint 'T.A' == 'T.B' written here}}
```

Comment:
This is definitely not ideal because these requirements should be diagnosed in source order, but I think I can fix that...